File: java/sdk/sdk-pay/src/main/java/com/novax/sdk/pay/withdraw/WithdrawOrdersRequest.java
package com.novax.sdk.pay.withdraw;

import com.novax.sdk.core.request.AbstractApiRequest;
import com.novax.sdk.core.request.HttpMethod;
import com.novax.sdk.core.request.TypeRef;
import com.novax.sdk.pay.model.WithdrawOrderResp;
import lombok.Builder;
import lombok.Getter;

import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

/** GET /pay/v3/withdraw/orders — query withdraw orders. */
@Getter
@Builder
public class WithdrawOrdersRequest extends AbstractApiRequest<List<WithdrawOrderResp>> {

    private final String withdrawOrderIds;
    private final Integer status;

    @Override public HttpMethod method() { return HttpMethod.GET; }
    @Override public String path() { return "/pay/v3/withdraw/orders"; }

    @Override public Map<String, ?> queryParams() {
        Map<String, Object> q = new LinkedHashMap<>();
        if (withdrawOrderIds != null) q.put("withdrawOrderIds", withdrawOrderIds);
        if (status != null) q.put("status", status);
        return q;
    }

    @Override public TypeRef<List<WithdrawOrderResp>> responseType() {
        return new TypeRef<>() {};
    }
}

File: python/novax_sdk/pay/withdraw/withdraw_orders_request.py
from dataclasses import dataclass
from typing import Any

from novax_sdk.pay.model import WithdrawOrderResp
from novax_sdk.request.api_request import ApiRequest, HttpMethod


@dataclass
class WithdrawOrdersRequest(ApiRequest):
    """GET /pay/v3/withdraw/orders"""

    withdraw_order_ids: str | None = None
    status: int | None = None

    @property
    def method(self) -> HttpMethod:
        return HttpMethod.GET

    @property
    def path(self) -> str:
        return "/pay/v3/withdraw/orders"

    def query_params(self) -> dict[str, Any]:
        return {k: v for k, v in {
            "withdrawOrderIds": self.withdraw_order_ids,
            "status": self.status,
        }.items() if v is not None}

    @property
    def response_type(self) -> type:
        return list[WithdrawOrderResp]
